packages/core/src/modules/kms/KeyManagementApi.ts

```diff
@@ -0,0 +1,371 @@
+        supportedBackends.push(kms.backend)
+      }
+    }
+
+    return supportedBackends
+  }
+
+  /**
+   * Create a key.
+   */
+  public async createKey<Type extends KmsCreateKeyType>(
+    options: WithBackend<KmsCreateKeyOptions<Type>>
+  ): Promise<KmsCreateKeyReturn<Type>> {
+    const { backend, ...kmsOptions } = parseWithErrorHandling(
+      zWithBackend(zKmsCreateKeyOptions),
+      options,
+      'Invalid options provided to createKey method'
+    )
+
+    const kms = this.getKms(this.agentContext, backend, {
+      operation: 'createKey',
+      type: options.type,
+    })
+
+    const key = await kms.createKey(this.agentContext, kmsOptions)
+    key.publicJwk.kid = key.keyId
+
+    this.agentContext.config.logger.debug(
+      `Created key ${getJwkHumanDescription(key.publicJwk)} with key id '${key.keyId}'`
+    )
+
+    return key
+  }
+
+  /**
+   * Create a key.
+   */
+  public async createKeyForSignatureAlgorithm(
+    options: WithBackend<KmsCreateKeyForSignatureAlgorithmOptions>
+  ): Promise<KmsCreateKeyReturn<KmsCreateKeyTypeAssymetric>> {
+    const { backend, algorithm, ...kmsOptions } = parseWithErrorHandling(
+      zWithBackend(zKmsCreateKeyForSignatureAlgorithmOptions),
+      options,
+      'Invalid options provided to createKeyForSignatureAlgorithm method'
+    )
+
+    const type = createKeyTypeForSigningAlgorithm(options.algorithm)
+    const kms = this.getKms(this.agentContext, backend, {
+      operation: 'createKey',
+      type,
+    })
+
+    // FIXME: do we want this?
+    // Ensure the kid is set to the keyId
+    const key = await kms.createKey(this.agentContext, {
+      ...kmsOptions,
+      type,
+    })
+    key.publicJwk.kid = key.keyId
+
+    return key
+  }
+
+  /**
+   * Sign using a key.
+   */
+  public async sign(options: WithBackend<KmsSignOptions>) {
+    const { backend, ...kmsOptions } = parseWithErrorHandling(
+      zWithBackend(zKmsSignOptions),
+      options,
+      'Invalid options provided to sign method'
+    )
+
+    const operation = {
+      operation: 'sign',
+      algorithm: options.algorithm,
+    } as const
+
+    const kms = backend
+      ? this.getKms(this.agentContext, backend, operation)
+      : (await this.getKmsForOperationAndKeyId(this.agentContext, options.keyId, operation)).kms
+    return await kms.sign(this.agentContext, kmsOptions)
+  }
+
+  /**
+   * Verify using a key.
+   */
+  public async verify(options: WithBackend<KmsVerifyOptions>) {
+    const { backend, ...kmsOptions } = parseWithErrorHandling(
+      zWithBackend(zKmsVerifyOptions),
+      options,
+      'Invalid options provided to verify method'
+    )
+
+    const operation = { operation: 'verify', algorithm: options.algorithm } as const
+    const kms =
+      backend || typeof options.key !== 'string'
+        ? this.getKms(this.agentContext, backend, operation)
+        : (await this.getKmsForOperationAndKeyId(this.agentContext, options.key, operation)).kms
```

Comment:
If options.key is an id I think it would be more natural to be called keyId (in KmsVerifyOptions)